core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -2426,18 +2428,43 @@ public void refreshExtensions() throws ExtensionRefreshException {
      */
     @Override
     public ACL getACL() {
-        return authorizationStrategy.getRootACL();
+        return getOverriddenAuthorizationStrategy().getRootACL();
     }
 
     /**
+     * {@link AuthorizationStrategy} configured in Global Security page.
+     *
+     * Use {@code getACL()} methods provided by items to get {@link ACL}s
+     * (e.g. {@link #getACL()}, {@link Job#getACL()}).
+     *
      * @return
      *      never null.
      */
+    @Nonnull
     public AuthorizationStrategy getAuthorizationStrategy() {
         return authorizationStrategy;
     }
 
     /**
+     * {@link AuthorizationStrategy} extended with extensions.
+     *
+     * This method is used by Jenkins internally and you never need to use this.
+     * You should use {@link #getAuthorizationStrategy()} to access the {@link AuthorizationStrategy}
+     * configured in Global Security page.
+     * And you should get {@link ACL}s with {@code getACL()} provided by target items,
+     * like {@link #getACL()}, {@link Job#getACL()} and so on.
+     *
+     * @return never {@code null}
+     * @since TODO
+     */
+    @Nonnull
+    public AuthorizationStrategy getOverriddenAuthorizationStrategy() {
```

Comment:
I tried that approach first and found that it breaks many plugins.
Plugins with implementations for `AuthorizationStrategy` often contains following codes:
```
if (Jenkins.getInstance().getAuthorizationStrategy() instanceof ItsOwnAuthorizationStrategy) {
    String configuredField = (ItsOwnAuthorizationStrategy(Jenkins.getInstance().getAuthorizationStrategy()).getSomeField();
    // codes using its configured fields...
}
```
Modifying the value from `getAuthorizationStrategy()` breaks those plugins.
e.g. `ProjectMatrixAuthorizationStrategy` no longer displays `AuthorizationMatrixProperty` in job configuration pages.

And I decided not to modify existing `getAuthorizationStrategy()`.